test/unit/view_model/access_control_test.rb

```diff
@@ -259,15 +276,15 @@ def test_visibility_from_root
       end
 
       refute_serializes(Tree1View, make_tree("visible", "invisible"))
-      assert_serializes(Tree1View, make_tree("visible_children", "invisible"))
+      assert_serializes(Tree1View, make_tree("visible_children", "visible"))
```

Comment:
This change, while technically the same test, seems to be the opposite spirit of the test. The change from `visible` to `visible_children` should be sufficient to change the `refute` to an `assert`. This implies that more changes were necessary.

Reply:
I think this was just wrong before. It asserted it serialized, meaning everything should've been visible.

Reply:
(saying `invisible` didn't break it: it still passed, it was just super confusing)

Reply:
There are two ways of reading this:
 - all expressions are controlling the access control
 - some expressions controlling access control and some talking to the reader about what the expected result is

I was doing the first, you're doing the second. The following test also is opposite. Pick one?

Reply:
I didn't realise they weren't all doing the second. My preference is the second (or at least a variation: _all_ expressions tell the reader what their result is, some of the values are also meaningful to the access control in order to cause that result as well) but I could be convinced otherwise?

Reply:
How about these changes?